transport/graphsync/initiating_test.go

```diff
@@ -0,0 +1,1325 @@
+	})
+
+	t.Run("pause", func(t *testing.T) {
+		th.Channel.SetInitiatorPaused(true)
+		th.Transport.ChannelUpdated(ctx, th.Channel.ChannelID(), th.UpdateRequest(true))
+		th.DtNet.AssertSentMessage(t, testharness.FakeSentMessage{PeerID: th.Channel.OtherPeer(), TransportID: "graphsync", Message: th.UpdateRequest(true)})
+		require.Len(t, th.Fgs.Pauses, 1)
+		require.Equal(t, th.Fgs.Pauses[0], request.ID())
+	})
+	t.Run("pause again", func(t *testing.T) {
+		th.Transport.ChannelUpdated(ctx, th.Channel.ChannelID(), th.UpdateRequest(true))
+		// should send message again
+		th.DtNet.AssertSentMessage(t, testharness.FakeSentMessage{PeerID: th.Channel.OtherPeer(), TransportID: "graphsync", Message: th.UpdateRequest(true)})
+		// should not pause again
+		require.Len(t, th.Fgs.Pauses, 1)
+	})
+	t.Run("resume", func(t *testing.T) {
+		th.Channel.SetInitiatorPaused(false)
+		th.Transport.ChannelUpdated(ctx, th.Channel.ChannelID(), th.UpdateRequest(false))
+		require.Len(t, th.Fgs.Resumes, 1)
+		resume := th.Fgs.Resumes[0]
+		require.Equal(t, request.ID(), resume.RequestID)
+		msg := resume.DTMessage(t)
+		require.Equal(t, msg, th.UpdateRequest(false))
+	})
+	t.Run("resume again", func(t *testing.T) {
+		th.Transport.ChannelUpdated(ctx, th.Channel.ChannelID(), th.UpdateRequest(false))
+		// should send message again
+		th.DtNet.AssertSentMessage(t, testharness.FakeSentMessage{PeerID: th.Channel.OtherPeer(), TransportID: "graphsync", Message: th.UpdateRequest(true)})
+		// should not resume again
+		require.Len(t, th.Fgs.Resumes, 1)
+	})
+
+	t.Run("restart request", func(t *testing.T) {
+		err := th.Transport.RestartChannel(ctx, th.Channel, th.RestartRequest(t))
+		require.NoError(t, err)
+		require.Len(t, th.DtNet.ProtectedPeers, 2)
+		require.Equal(t, th.DtNet.ProtectedPeers[1], testharness.TaggedPeer{th.Channel.OtherPeer(), th.Channel.ChannelID().String()})
+		require.Len(t, th.DtNet.ConnectWithRetryAttempts, 1)
+		require.Equal(t, th.DtNet.ConnectWithRetryAttempts[0], testharness.ConnectWithRetryAttempt{th.Channel.OtherPeer(), "graphsync"})
+		th.DtNet.AssertSentMessage(t, testharness.FakeSentMessage{PeerID: th.Channel.OtherPeer(), TransportID: "graphsync", Message: th.NewRequest(t)})
+	})
+
+	t.Run("complete request", func(t *testing.T) {
+		th.ResponseCompletedListener(request, graphsync.RequestCompletedFull)
+		select {
+		case <-th.CompletedResponses:
+		case <-ctx.Done():
+			t.Fatalf("did not complete request")
+		}
+		th.Events.AssertTransportEvent(t, th.Channel.ChannelID(), datatransfer.TransportCompletedTransfer{Success: true})
+	})
+
+	t.Run("cleanup request", func(t *testing.T) {
+		th.Transport.CleanupChannel(th.Channel.ChannelID())
+		require.Len(t, th.DtNet.UnprotectedPeers, 1)
+		require.Equal(t, th.DtNet.UnprotectedPeers[0], testharness.TaggedPeer{th.Channel.OtherPeer(), th.Channel.ChannelID().String()})
+	})
+}
+
+/*	"gs outgoing request with recognized dt push channel will record incoming blocks": {
```

Comment:
are there still pending tests in this comment block to be ported?

Reply:
no they should go. finishing those out right now.